lib/frrstr.h

```diff
@@ -180,6 +180,12 @@ const char *frrstr_skip_over_char(const char *s, int skipc);
  */
 const char *frrstr_back_to_char(const char *s, int toc);
 
+/*
+ * Print time as a string, but trim the newline char.
+ * For JSON outputs, newline char is not expected.
+ */
+const char *frrstr_time(time_t *tbuf, char *buf);
```

Comment:
a little confused about the need for this - we have lib/monotime.h:time_to_string() already? imo it'd be clearer to have one place to do the "normal" and "json" variations.

Reply:
do you suggest extending time_to_string() to filter out newline or two separate functions?

Reply:
we even have `timestamp_string()` :smile:

Reply:
right - I'd prefer having one path into ctime_r() with our various wrappers, so it's a little easier to figure out what we think is safe.
maybe we could have time_to_str_json(), which calls our wrapper and then fixes the output buffer?

Reply:
What about I drop `timestamp_string()`, `frrstr_time()`, and extend `time_to_string()` to keep newline or strip it?

Reply:
> maybe we could have time_to_str_json(), which calls our wrapper and then fixes the output buffer?

That sounds good, but now the question is which wrapper `timestamp_string()` or `time_to_string()`? :smile: Do you agree to remove one and use one instead of two?

Reply:
ah, yes, I didn't realize bgpd had a version also!  I think it would be best to have one version, in lib/, rather than per-daemon versions. and then the lib could also have a xxx_json() version, that strips the newline (if present)

Reply:
To summarize is this fine?:
* keep `time_to_string()` (lib/monotime.h);
* create `time_to_string_json()` wrapper for `time_to_string()`;
* remove `timestamp_string()` - replace with previous functions.

Reply:
yes, that sounds excellent to me

Reply:
Should be fixed according what we talked \m/